Make Array-creation behavior of Paginable more predictable

akihikodaki opened the pull request:
`Paginable.paginate_by_id` usually returns `ActiveRecord::Relation`, but it returns an `Array` if `min_id` option is present. The behavior caused problems fixed with the following commits:
- 552e886b648faa2a2229d86c7fd9abc8bb5ff99c
- b63ede5005d33b52266650ec716d345f166e2df0
- 64ef37b89de806f49cc59e011aa0ee2039c82c46

To prevent from recurring similar problems, this commit introduces two changes:
- The scope now always returns an `Array` whether min_id option is present or not.
- The scope is renamed to `to_a_paginated_by_id` to clarify it returns an `Array`.

akihikodaki committed:
Make Array-creation behavior of Paginable more predictable

Paginable.paginate_by_id usually returns ActiveRecord::Relation, but it
returns an Array if min_id option is present. The behavior caused problems
fixed with the following commits:
- 552e886b648faa2a2229d86c7fd9abc8bb5ff99c
- b63ede5005d33b52266650ec716d345f166e2df0
- 64ef37b89de806f49cc59e011aa0ee2039c82c46

To prevent from recurring similar problems, this commit introduces two
changes:
- The scope now always returns an Array whether min_id option is present
  or not.
- The scope is renamed to to_a_paginated_by_id to clarify it returns an
  Array.

ThibG commented:
hmmm, a scope should return an `ActiveRecord::Relation`, not an array, so while I'm fine with a `to_a_paginated_by_id` class method, I think the scope itself should return a relation. In the `min_id` case, I guess it was a bug and it should call [`reverse_order`](https://apidock.com/rails/ActiveRecord/QueryMethods/reverse_order) rather than `reverse`

akihikodaki commented:
> hmmm, a scope should return an `ActiveRecord::Relation`, not an array, so while I'm fine with a `to_a_paginated_by_id` class method, I think the scope itself should return a relation. In the `min_id` case, I guess it was a bug and it should call [`reverse_order`](https://apidock.com/rails/ActiveRecord/QueryMethods/reverse_order) rather than `reverse`

`reverse_order` changes the behavior in an unintended way. If `min_id` is provided, it is expected to return the records with smallest IDs greater than `min_id`. However, `reverse_order` makes it return the records with the greatest IDs greater than `min_id`.
It should be a class method instead of a scope.

akihikodaki committed:
Transform Paginable.to_a_paginated_by_id from a scope to a class method

https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope
> The method is intended to return an ActiveRecord::Relation object, which
> is composable with other scopes.

Paginable.to_a_paginated_by_id returns an Array and is not appropriate
as a scope.

akihikodaki commented:
95ad022e44c22b11f142c195d3c6df7f668ab951: Transform `Paginable.to_a_paginated_by_id` from a scope to a class method.
